src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,132 @@
 package core.basesyntax;
 
-public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+import java.util.NoSuchElementException;
+
+public class ArrayList<T> implements MyList<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double MULTIPLIER = 1.5;
+    private static final String MESSAGE_OUT_OF_BOUNDS = "The index passed to any of "
+            + "the methods is invalid";
+    private static final String MESSAGE_NO_ELEMENT = "There is no such element";
+
+    private T[] elements;
+    private int size = 0;
 
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
     }
 
     @Override
-    public void add(T value, int index) {
-
+    public boolean add(T value) {
+        increaseCapacityIfFull();
+        elements[size++] = (T) value;
+        return true;
     }
 
     @Override
-    public void addAll(List<T> list) {
+    public void add(T value, int index) {
+        validateIndexForAdd(index);
+        increaseCapacityIfFull();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
+    }
 
+    @Override
+    public void addAll(MyList<T> list) {
+        while (size + list.size() > elements.length) {
+            grow();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i)); // Додаємо елементи з переданого списку
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validateIndex(index);
+        return (T) elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        validateIndex(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        validateIndex(index);
+        T oldValue = (T) elements[index];
+        System.arraycopy(elements,index + 1, elements, index, size - index - 1);
+        elements[--size] = null;
+        return oldValue;
     }
 
     @Override
-    public T remove(T element) {
-        return null;
+    public T remove(Object element) {
+        int indexOfElement = indexOf(element);
+        if (indexOfElement == -1) {
+            throw new NoSuchElementException(MESSAGE_NO_ELEMENT);
+        } else {
+            System.arraycopy(elements, indexOfElement + 1, elements, indexOfElement,
+                    size - indexOfElement - 1);
+            elements[--size] = null;
+            return (T) element;
```

Comment:
In the 'remove(Object element)' method, the return statement 'return (T) element;' is incorrect. It should return the old value that was removed, similar to the 'remove(int index)' method. This is a logical error.